File: PointOfSale/POS.WebApp/web/src/models/interfaces/product/ProductViewModel.ts
import { CommonStatus } from "../../enums/CommonStatus";
import { CategoryLiteViewModel } from "../category/CategoryLiteViewModel";

export interface ProductViewModel {
    id:string;
    code:string;
    name:string;
    description?:string;
    imagePath?:string;
    status:CommonStatus;
    statusDescription:string;
    category:CategoryLiteViewModel;
}

File: PointOfSale/POS.Core/Models/ViewModels/Product/ProductViewModel.cs
using POS.Core.Models.ViewModels.Category;
using POS.Domain.Models.Enums;

namespace POS.Core.Models.ViewModels.Product
{
    public class ProductViewModel
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public CommonStatus Status { get; set; }
        public string StatusDescription => Status.ToString();

        public CategoryLiteViewModel Category { get; set; }
    }
}
